lib/mix/lib/mix.ex

```diff
@@ -870,6 +877,13 @@ defmodule Mix do
 
         try do
           first_build? = not File.dir?(build_dir)
+
+          restore_dir = System.get_env("MIX_INSTALL_RESTORE_PROJECT_DIR")
+
+          if first_build? and restore_dir != nil and not force? do
+            File.cp_r!(restore_dir, install_dir)
```

Comment:
Do we want it to fail if we cannot copy? Maybe ignoring it is fine?

```suggestion
            File.cp_r(restore_dir, install_dir)
```